ParquetClassLibrary/AssemblyInfo.cs

```diff
@@ -62,11 +62,42 @@ public struct AssemblyInfo
         /// </summary>
         public static readonly Range<EntityID> CollectibleIDs = new Range<EntityID>(40000, 49000);
 
+        /// <summary>
+        /// The largest <see cref="Range{EntityID}.Maximum"/> defined in <see cref="AssemblyInfo"/>,
+        /// excluding <see cref="ItemIDs"/>.
+        /// </summary>
+        private static readonly int IDsMaxExcludingItems = typeof(AssemblyInfo).GetFields()
```

Comment:
It looks like the purpose of this is to get the maximum ID when you consider FloorIDs, BlockIDs, FurnishingIDs, and CollectibleIDs. It seems to be doing reflectiony stuff, asking for fields.

Is there a reason we can't pull from the ParquetIDs list, instead of pulling things via reflection? This is in AssemblyInfo, so it'd make sense if it had to be done in this way, but at first glance it seems unusual.

If you're allowed to pull in from ParquetIDs, consider;

ParquetIDs.Select(x => x.Maximum).Max();

Reply:
Thank you for these thoughts!

So, the reason I am not pulling directly from the ParquetIDs list is because I anticipate using EntityID to identify other non-parquet, non-item things: crafting recipes, room recipes, etc.

The reason I am using reflection rather than specifying the fields myself is to avoid having to update a manually-entered list every time I add a new Range<EntityID>.

I was thinking that perhaps once the era of rapid development is over it might make sense to replace this with a manually-specified list in the interests of clarity.  But while I'm still iterating on my ideas rapidly I'm worried about setting up a bunch of CSV files with invalid IDs in them.

Thoughts?

Reply:
Seems good; that's the primary strength of this particular reflection pattern. In that case, this approach seems fine, especially because this is a static readonly that only gets computed once. I get a little alerted whenever I see reflection used, but this isn't going to be something that's taxing on the system.

I don't immediately know a more effective way of approaching this particular Linq statement, so it seems good to me!